### PR TITLE
2 get topics

### PR DESCRIPTION
**Pull request by Sir1ys**

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+beforeEach(() => seed(data));
+
+afterAll(() => db.end());
+
+describe("/wrong-request", () => {
+  // ERROR TESTING
+  test("GET:400 when request was made to wrong URL", () => {
+    return request(app)
+      .get("/wrong-request")
+      .expect(400)
```
Comment (B7Bprog) [inline __tests__/index.test.js]: This should not be a bad request 400 here. Think about what it should be where we could have that path, but we don't have it yet so can not be found.

File: __tests__/index.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+describe("/api/topics", () => {
+  test("GET:200 sends an array of topics to the client", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const topics = body.topics;
+
+        expect(topics).toHaveLength(3);
+        topics.forEach((topic) => {
```
Comment (B7Bprog) [inline __tests__/index.test.js:33]: This will work, but I recommend the use of jest matchers.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const express = require("express");
+const { getTopics } = require("./controllers/topics.controller");
+
+const app = express();
+
+app.get("/api/topics", getTopics);
+
+app.all("*", (req, res) => {
+  throw new Error("Bad request");
```
Comment (B7Bprog) [inline app.js]: You could send a response from here.

File: app.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    });
+  }
+
+  next(err);
+});
+
+// Custom Errors handling
+app.use((err, req, res, next) => {
+  if (err.status) {
+    console.log("sss");
```
Comment (B7Bprog) [inline app.js]: Please remove all debugging logs from final code.

File: app.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    res.status(err.status).send({
+      msg: err.msg,
+    });
+  }
```
Comment (B7Bprog) [inline app.js:34]: Please include an error handling middleware for internal server errors.

File: models/topics.model.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const db = require("../db/connection");
+
+exports.fetchTopics = () => {
+  const query = `SELECT * FROM topics`;
+
+  return db
+    .query(query)
+    .then(({ rows }) =>
+      rows.length === 0
+        ? Promise.reject({ msg: "Bad request", status: 400 })
```
Comment (B7Bprog) [inline models/topics.model.js]: An empty array of topics should be just fine. It shouldn't be an error.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Sasha, please check my comments and refactor your code, then send us another PR with it.

**Review by elijaharley — APPROVED** [review]
To make your tests more robust I'd consider using toMatchObject syntax or expect(topic).toHaveProperty('slug', expect.any(String)) instead of typeof. otherwise good stuff, you can merge when ready.

**Pull request merged**